src/System.Linq.Expressions/tests/ReadOnlyCollectionBuilder/ReadOnlyCollectionBuilder.Tests.Misc.cs

```diff
@@ -0,0 +1,560 @@
+                }
+            }
+
+            public void NonGenericIListInsertValidations(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                IList _ilist = list;
+                int[] bad = new int[] { items.Length + 1, items.Length + 2, int.MaxValue, -1, -2, int.MinValue };
+                for (int i = 0; i < bad.Length; i++)
+                {
+                    Assert.Throws<ArgumentOutOfRangeException>(() => _ilist.Insert(bad[i], items[0])); //"ArgumentOutOfRangeException expected."
+                }
+
+                Assert.Throws<ArgumentException>(() => _ilist.Insert(0, new LinkedListNode<string>("blargh"))); //"ArgumentException expected."
+            }
+
+            #endregion
+
+            #region Contains
+
+            public void BasicContains(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+
+                for (int i = 0; i < items.Length; i++)
+                {
+                    Assert.True(list.Contains(items[i])); //"Should contain item."
+                }
+            }
+
+            public void NonExistingValues(T[] itemsX, T[] itemsY)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(itemsX);
+
+                for (int i = 0; i < itemsY.Length; i++)
+                {
+                    Assert.False(list.Contains(itemsY[i])); //"Should not contain item"
+                }
+            }
+
+            public void RemovedValues(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                for (int i = 0; i < items.Length; i++)
+                {
+                    list.Remove(items[i]);
+                    Assert.False(list.Contains(items[i])); //"Should not contain item"
+                }
+            }
+
+            public void AddRemoveValues(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                for (int i = 0; i < items.Length; i++)
+                {
+                    list.Add(items[i]);
+                    list.Remove(items[i]);
+                    list.Add(items[i]);
+                    Assert.True(list.Contains(items[i])); //"Should contain item."
+                }
+            }
+
+            public void MultipleValues(T[] items, int times)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+
+                for (int i = 0; i < times; i++)
+                {
+                    list.Add(items[items.Length / 2]);
+                }
+
+                for (int i = 0; i < times + 1; i++)
+                {
+                    Assert.True(list.Contains(items[items.Length / 2])); //"Should contain item."
+                    list.Remove(items[items.Length / 2]);
```

Comment:
You can improve this slightly: `Assert.True(list.Remove(items[items.Length / 2])`